### DISCUSSION TITLE
linkToInput в расчётной в цикле

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#130] [ОШБК!] [answered]
![image](https://user-images.githubusercontent.com/103171127/168228495-dad94012-6044-4a3d-b0f5-a7b9140f04b0.png)
![image](https://user-images.githubusercontent.com/103171127/168228768-be86a447-bf9f-4e98-877c-148f10289333.png)![image](https://user-images.githubusercontent.com/103171127/168228599-58b1efec-7171-4c4d-9a3c-d54564b87f28.png)

**Comment by erpic-ru** [comment, marked as answer]
можно попробовать убрать var и cycle, либо заменить $#nci на #id, если мы говорим о соседней ячейке

UPD: это ОШБК! исправлено в 2.4.42.0

Должно работать вот так:

```
= : linkToInput(title: "test"; code: $code; refresh: true; button: "Save")

     ```code:totum
     =: set(table: $#ntn; field: 'test' = $#input; where: 'id' = #id; log: true)
     ```
```

**Reply by ttm-alexeypolunin** [reply]
Если бы это была таблица внутри цикла, то тогда нужен был бы `cycle`, а так таблица циклов аналогична простой.

Проверка показала, что `linkToInput` получает при вызове `#id` и `$#ntn` и `$#nci` — поэтому если бы это было внутри цикла, то нужно было бы использовать `cycle: $#nci`. Но в самой таблице циклов `$#nci` нет, потому что это `#id`.

Текст ошибки дурацкий — из него непонятно что речь идет про `$#nci`.

**Reply by Makhmut2445** [reply]
Эта и есть таблица внутри цикла. Я же написал в названии что действие происходит в расчётной в цикле. Как использовать var внутри?

**Reply by ttm-alexeypolunin** [reply]
ага, вижу, что поломалось

**Reply by Makhmut2445** [reply]
И что же? Что не так делаю?

**Reply by ttm-alexeypolunin** [reply]
Если внутри цикла, то должно работать как у вас написано — мы полезли изучать смутное сообщение об ошибке с 0 циклом и поняли что там ошибка в ядре.

Напишу как разберемся

**Reply by Makhmut2445** [reply]
Хорошо

**Reply by Makhmut2445** [reply]
![image](https://user-images.githubusercontent.com/103171127/168252763-514f849f-1b89-4145-ac3a-0b386ad55fb2.png)
Есть ещё такая ошибка, при каждом переключении между пользователями....

**Reply by ttm-alexeypolunin** [reply]
Вобщем должно работать даже вот так:

```
= : linkToInput(title: "test"; code: $code; refresh: true; button: "Save")

     ```code:totum
     =: set(table: $#ntn; field: 'test' = $#input; where: 'id' = #id; log: true)
     ```
```

`cycle` — должен автоматически подставляться (с этой автоподстановкой и была ошибка).

Обновление будет сег вечером.

**Reply by ttm-alexeypolunin** [reply]
**fix in 2.4.42**
